### PR TITLE
Travis testing for Piranha dependency

### PR DESCRIPTION
**Pull request by Sumith1896**
We'll need to run the tests `WITH_PIRANHA=yes` as, at least for now, `Polynomial` will depend on `Piranha`
cc @certik @isuruf @shivamvats

### CODE REVIEW

**Comment by certik** [comment]
You should skip building Piranhas tests and benchmarks and you should
compile it in Release mode (not sure if that's a default).

**Comment by Sumith1896** [comment]
It is asking for the latest CMake, how do I install that?
Since there is no sudo access

**Comment by isuruf** [comment]
You can install it into `$our_install_dir` by specifying `--prefix`.
Even though `cmake 2.8.10` is required in the CMakeLists.txt file, I think all features used are available in `cmake 2.8.7`. @bluescarni, is 2.8.10 necessary?

**Comment by bluescarni** [comment]
@isuruf my recollection is that I kind of just picked an arbitrary CMake version which probably was reasonably recent at the time. I am not sure whether 2.8.7 would work or not to be honest.

**Comment by isuruf** [comment]
@bluescarni, I just checked and 2.8.7 is enough for piranha. I posted https://github.com/bluescarni/piranha/pull/11.

@Sumith1896, you need to call cmake using `$our_install_dir/bin/cmake` to use the latest cmake you installed, otherwise old version is picked up.

To install piranha in `$our_install_dir` you can't give `--prefix`. instead you could give `-DCMAKE_INSTALL_PREFIX=$our_install_dir`

File: .travis.yml [hunk 1]
```diff
@@ -60,6 +60,17 @@ matrix:
...
+      compiler: gcc
+      os: linux
+      addons:
+        apt:
+          sources:
+          - ubuntu-toolchain-r-test
+          - boost-latest
+          packages:
+          - libboost-serialization1.55-dev
+          - g++-4.8
```
Comment (isuruf) [inline .travis.yml]: You have to add `libgmp-dev` and `libmpfr-dev` here as well as this entry is overriding the above project-wide entry.

**Comment by Sumith1896** [comment]
@certik This is ready. Could you have a look?

**Comment by certik** [comment]
I think that this looks good, great job!

**Comment by Sumith1896** [comment]
Thanks for merging :)

**Pull request merged**